### PR TITLE
Added tests for multi_platform_controller_cleanup_failures metric

### PR DESCRIPTION
**Pull request by glevi-rh**
## What

* Added a new test case to check if the `multi_platform_controller_cleanup_failures` counter is correctly incremented when a cleanup task fails.
* Added a corresponding test to confirm the counter is not incremented when a cleanup task succeeds.

## Why

* The `multi_platform_controller_cleanup_failures` metric consistently reports a value of zero, so these tests were created to verify that the metric is being handled correctly and that there is no existing bug preventing it from incrementing.
* This also addresses a gap in our test coverage, as we didn't have a test scenario for a cleanup failure.

### CODE REVIEW

File: pkg/reconciler/taskrun/taskrun_test.go [hunk 1]
```diff
@@ -304,4 +310,153 @@ var _ = Describe("TaskRun Reconciler General Tests", func() {
...
+		getCleanupFailuresMetric := func(platform string) float64 {
+			metricDto := &dto.Metric{}
+			var pmetrics *mpcmetrics.PlatformMetrics
+			mpcmetrics.HandleMetrics(platform, func(metrics *mpcmetrics.PlatformMetrics) {
+				pmetrics = metrics
+			})
+			Expect(pmetrics).ShouldNot(BeNil())
+			Expect(pmetrics.CleanupFailures.Write(metricDto)).ShouldNot(HaveOccurred())
+			return metricDto.GetCounter().GetValue()
+		}
```
Comment (meyrevived) [inline pkg/reconciler/taskrun/taskrun_test.go:336]: Is this a function that serves only as a helper, and the assertion lines here just making sure every functionality in it works properly? 
Or is it a crucial part of the test scenario and is expected to behave differently in every different test case?

If this is purely a helper function, these usually belong outside of the ginkgo scope, [like this helper function](https://github.com/konflux-ci/multi-platform-controller/blob/53a0feceec69dc434b84cbd40a8a6ce506c7cff7/pkg/reconciler/taskrun/hostupdate_test.go#L37). 
Specifically for taskrun_test.go, which has been and can still become a gigantic and hard to read file, helper functions belong in [taskrun_helpers_test.go](https://github.com/konflux-ci/multi-platform-controller/blob/main/pkg/reconciler/taskrun/taskrun_helpers_test.go).

If this is part of the test, I personally would still place it outside of the ginkgo scope.

File: pkg/reconciler/taskrun/taskrun_test.go [hunk 2]
```diff
@@ -304,4 +310,153 @@ var _ = Describe("TaskRun Reconciler General Tests", func() {
...
+
+			// Reconcile the cleanup task - this should increment the CleanupFailures metric
+			_, err = reconciler.Reconcile(ctx, reconcile.Request{
+				NamespacedName: types.NamespacedName{Namespace: cleanupTask.Namespace, Name: cleanupTask.Name},
+			})
+			Expect(err).ShouldNot(HaveOccurred())
+
+			// Verify the CleanupFailures metric incremented
+			newFailures := getCleanupFailuresMetric(platform)
+			Expect(newFailures).Should(Equal(initialFailures + 1))
```
Comment (meyrevived) [inline pkg/reconciler/taskrun/taskrun_test.go:399]: In this spirit of @sadlerap  - this can be a one-liner
```suggestion
			Expect(getCleanupFailuresMetric(platform)).Should(Equal(initialFailures + 1))
```

File: pkg/reconciler/taskrun/taskrun_test.go [hunk 3]
```diff
@@ -304,4 +310,153 @@ var _ = Describe("TaskRun Reconciler General Tests", func() {
...
+		})
+
+		It("should NOT increment CleanupFailures metric when cleanup task succeeds", func(ctx SpecContext) {
```
Comment (meyrevived) [inline pkg/reconciler/taskrun/taskrun_test.go:402]: Usually positive test cases (happy paths, where everything is expected to work according to the logic in the code) are the first in a test suite. This is because the most important and basic question to ask is "Does it work as intended, when given everything it needs to succeed?". Then you have tests that ask "Does it fail when it supposed to (the way it's expected to fail)?" and then you move on to tests for edge cases that say "Let's try to mess around it, see if it breaks" and so on.

**Review by meyrevived — CHANGES_REQUESTED** [review]
Nice clean work - some fixes are required to make this more standardized to testing work in general and to some team-specific preferences.
